src/tdi_table_data.cpp

```diff
@@ -161,11 +184,25 @@ tdi_status_t TableData::getParent(const tdi::Learn ** /*learn*/) const {
   LOG_ERROR("%s:%d Not supported", __func__, __LINE__);
   return TDI_NOT_SUPPORTED;
 }
+
 tdi_status_t TableData::isActive(const tdi_id_t &field_id,
                                  bool *is_active) const {
-  *is_active =
-      std::find(active_fields_.begin(), active_fields_.end(), field_id) !=
-      active_fields_.end();
+  auto it = this->removed_one_ofs_.find(field_id);
```

Comment:
can you add comments for removed_one_ofs_ in terms all_fields_set_ what happens for all_fields_set_ is True? Should we check the all_field_set_ first or not?